roles/keycloak/tasks/main.yml

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) 2022 VEXXHOST, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Create a Keycloak namespace
+  kubernetes.core.k8s:
+    state: present
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: "{{ keycloak_helm_release_namespace }}"
+
+- name: Get the Kuberentes service for Percona XtraDB Cluster
+  run_once: true
+  kubernetes.core.k8s_info:
+    kind: Service
+    name: "{{ openstack_helm_endpoints.oslo_db.hosts.default }}"
+    namespace: openstack
+  register: _pxc_service
+
+- name: Install Kubernetes python package
+  ansible.builtin.pip:
+    name: PyMySQL
+
+- name: Create Keycloak database
+  run_once: true
+  community.mysql.mysql_db:
+    login_host: "{{ _pxc_service.resources[0].spec.clusterIP }}"
+    login_user: root
+    login_password: "{{ openstack_helm_endpoints.oslo_db.auth.admin.password }}"
+    name: "{{ keycloak_database_name }}"
+
+- name: Check if the Keycloak database secret exists
+  run_once: true
+  kubernetes.core.k8s_info:
+    api_version: v1
+    kind: Secret
+    name: keycloak-db-credentials
+    namespace: "{{ keycloak_helm_release_namespace }}"
+  register: _keycloak_db_secret
+  ignore_errors: true
+
+- name: Create a Keycloak database secret
+  when: ( _keycloak_db_secret.resources | length==0 )
+  kubernetes.core.k8s:
+    state: present
+    definition:
+      apiVersion: v1
+      stringData:
+        username: keycloak
+        password: "{{ lookup('password', '/dev/null chars=ascii_lowercase,ascii_uppercase,digits length=32') }}"
+      kind: Secret
+      metadata:
+        name: keycloak-db-credentials
+        namespace: "{{ keycloak_helm_release_namespace }}"
+      type: Opaque
+
+- name: Grab Keycloak database secret
+  kubernetes.core.k8s_info:
+    api_version: v1
+    kind: Secret
+    name: keycloak-db-credentials
+    namespace: "{{ keycloak_helm_release_namespace }}"
+  register: _keycloak_db_secret
+
+- name: Create a Keycloak user
+  community.mysql.mysql_user:
+    login_host: "{{ _pxc_service.resources[0].spec.clusterIP }}"
+    login_user: root
+    login_password: "{{ openstack_helm_endpoints.oslo_db.auth.admin.password }}"
+    name: "{{ _keycloak_db_secret.resources[0]['data']['username'] | b64decode }}"
+    password: "{{ _keycloak_db_secret.resources[0]['data']['password'] | b64decode }}"
```

Comment:
This would be simply getting it from the defaults, where the username would default to `keycloak` and the password would be requested by the user.